src/base-handler.ts

```diff
@@ -0,0 +1,177 @@
+import {
+    ArrayConcat,
+    getPrototypeOf,
+    getOwnPropertyNames,
+    getOwnPropertySymbols,
+    getOwnPropertyDescriptor,
+    isUndefined,
+    isExtensible,
+    hasOwnProperty,
+    unwrap,
+    ObjectDefineProperty,
+    preventExtensions,
+    isArray,
+    isObject,
+    freeze,
+} from './shared';
+import { ReactiveMembrane } from './reactive-membrane';
+
+export type ReactiveMembraneShadowTarget = object | any[];
```

Comment:
should there be a third type be
_export type ReactiveMembraneShadowTarget = object | any[] | undefined;_
Looking at line 28, if neither of the branches are satisfied then the default value of shadowTarget remains undefined